src/Confluent.Kafka/Exceptions/TopicPartitionException.cs

```diff
@@ -24,7 +24,7 @@ namespace Confluent.Kafka
     /// <summary>
     ///     Represents an error that occured during a Consumer.Position request.
     /// </summary>
-    public class TopicPartitionException : Exception
+    public class TopicPartitionException : KafkaException
```

Comment:
two things:

1. like the admin client exceptions, it's best if this derives from `KafkaException` but we need a new `ErrorCode` for it, unknown isn't good enough.

2. left as todo: we should have exception types that drive from this e.g. `ConsumerPauseException`.